lib/eventFactory.js

```diff
@@ -406,6 +406,13 @@ function createBuilds(config) {
 
                     buildConfig.configPipelineSha = pipelineConfig.configPipelineSha;
 
+                    const { annotations, freezeWindows } = j.permutations[0];
+                    const isVirtualJob = annotations ? annotations['screwdriver.cd/virtualJob'] === true : false;
+                    const hasFreezeWindows = freezeWindows ? freezeWindows.length > 0 : false;
+
+                    // Bypass execution of the build if the job is virtual
+                    buildConfig.start = !isVirtualJob || (isVirtualJob && hasFreezeWindows);
```

Comment:
```suggestion
                    buildConfig.start = !isVirtualJob || hasFreezeWindows;
```
`isVirtualJob` will always evaluate to true for the second  half of the or condition

Reply:
Done